crates/nu-command/src/filesystem/watch.rs

```diff
@@ -95,20 +115,57 @@ impl Command for Watch {
 
         let quiet = call.has_flag(engine_state, stack, "quiet")?;
 
-        let debounce_duration_flag: Option<Spanned<i64>> =
+        let debounce_duration_flag_ms: Option<Spanned<i64>> =
             call.get_flag(engine_state, stack, "debounce-ms")?;
-        let debounce_duration = match debounce_duration_flag {
-            Some(val) => match u64::try_from(val.item) {
-                Ok(val) => Duration::from_millis(val),
-                Err(_) => {
+
+        let debounce_duration_flag: Option<Spanned<Value>> =
+            call.get_flag(engine_state, stack, "debounce")?;
+
+        let debounce_duration: Duration;
+        match (debounce_duration_flag, debounce_duration_flag_ms) {
+            (None, None) => debounce_duration = DEFAULT_WATCH_DEBOUNCE_DURATION,
+            (Some(l), Some(r)) => {
+                return Err(ShellError::IncompatibleParameters {
+                    left_message: "Here".to_string(),
+                    left_span: l.span,
+                    right_message: "and here".to_string(),
+                    right_span: r.span,
+                });
+            }
+            (None, Some(val)) => {
+                debounce_duration = match u64::try_from(val.item) {
+                    Ok(v) => Duration::from_millis(v),
+                    Err(_) => {
+                        return Err(ShellError::TypeMismatch {
+                            err_message: "Debounce duration is invalid".to_string(),
+                            span: val.span,
+                        });
+                    }
+                }
+            }
+            (Some(v), None) => match v.item {
+                dur @ Value::Duration { val, .. } => {
+                    debounce_duration = match u64::try_from(val) {
+                        Ok(d) => Duration::from_nanos(d),
+                        Err(_) => {
+                            return Err(ShellError::TypeMismatch {
+                                err_message: "Debounce duration is invalid".to_string(),
+                                span: dur.span(),
+                            });
+                        }
+                    };
+                }
+                // This should be unreachable, as the parser will fail
+                // before this is ever called. I will leave it in anyways
+                // so we can handle it gracefully once it never arrives
+                any => {
```

Comment:
I think we can just do this here:

```suggestion
            (Some(v), None) => {
                let Value::Duration { val, ..} = v else {
                    return Err(ShellError::TypeMismatch {
                        err_message: "Debounce duration must be a duration".to_string(),
                        span: dur.span(),
                    });
                }
                val
```

this suggestion won't apply cleanly since I can't do a suggestion over unchanged lines, but the `any` match branch would have to be manually removed as well

Reply:
fair, that is kind of repetitive. In my most recent commit, I removed the any branch and refactored this according to your suggestion. I also use `v.item.span()` instead of `v.span` due to the [deprecation notice](https://github.com/lucascherzer/nushell/blob/c92711c3f4fe59d608f553ed65108ee260cd1679/crates/nu-protocol/src/value/mod.rs#L94-L97)

Reply:
So do you want this to use `v.span` instead? What about the deprecation notice - how is this planned to go in the future?